pkg/ttnpb/field_mask_validation.go

```diff
@@ -0,0 +1,190 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ttnpb
+
+var isEndDeviceReadFieldPaths = []string{
+	"application_server_address",
+	"attributes",
+	"created_at",
+	"description",
+	"ids",
+	"ids.application_ids",
+	"ids.application_ids.application_id",
+	"ids.dev_eui",
+	"ids.device_id",
+	"ids.join_eui",
+	"join_server_address",
+	"locations",
+	"name",
+	"network_server_address",
+	"service_profile_id",
+	"updated_at",
+	"version_ids",
+	"version_ids.brand_id",
+	"version_ids.firmware_version",
+	"version_ids.hardware_version",
+	"version_ids.model_id",
+}
+
+var isEndDeviceWriteFieldPaths = []string{
+	"application_server_address",
+	"attributes",
+	"description",
+	"ids",
+	"ids.dev_eui",
+	"ids.join_eui",
+	"join_server_address",
+	"locations",
+	"name",
+	"network_server_address",
+	"service_profile_id",
+	"version_ids",
+	"version_ids.brand_id",
+	"version_ids.firmware_version",
+	"version_ids.hardware_version",
+	"version_ids.model_id",
+}
+
+var allowedFieldMaskPathsForRPC = map[string][]string{
+	// Applications:
+	"/ttn.lorawan.v3.ApplicationRegistry/Get":                 ApplicationFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationRegistry/List":                ApplicationFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationRegistry/Update":              ApplicationFieldPathsNested,
+	"/ttn.lorawan.v3.EntityRegistrySearch/SearchApplications": ApplicationFieldPathsNested,
+
+	// Application Webhooks:
+	"/ttn.lorawan.v3.ApplicationWebhookRegistry/Get":  ApplicationWebhookFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationWebhookRegistry/List": ApplicationWebhookFieldPathsNested,
+	"/ttn.lorawan.v3.ApplicationWebhookRegistry/Set":  ApplicationWebhookFieldPathsNested,
+
+	// Application Links:
+	"/ttn.lorawan.v3.As/GetLink": ApplicationLinkFieldPathsNested,
+	"/ttn.lorawan.v3.As/SetLink": ApplicationLinkFieldPathsNested,
+
+	// Clients:
+	"/ttn.lorawan.v3.ClientRegistry/Get":                 ClientFieldPathsNested,
+	"/ttn.lorawan.v3.ClientRegistry/List":                ClientFieldPathsNested,
+	"/ttn.lorawan.v3.ClientRegistry/Update":              ClientFieldPathsNested,
+	"/ttn.lorawan.v3.EntityRegistrySearch/SearchClients": ClientFieldPathsNested,
+
+	// End Devices:
+	"/ttn.lorawan.v3.AsEndDeviceRegistry/Get":                  EndDeviceFieldPathsNested,
```

Comment:
@johanstokking can you add allowed set of fieldpaths for AS methods?

Reply:
Did that